src/nyc_trees/apps/census_admin/views.py

```diff
@@ -3,33 +3,71 @@
 from __future__ import unicode_literals
 from __future__ import division
 
+from collections import OrderedDict
+import json
 
-def start_admin_users_job(request):
-    # TODO: implement
-    # TODO: implement
-    pass
+from django.contrib.gis.geos import MultiPolygon, Polygon
+from django.db import transaction
 
+from apps.core.models import Group
 
-def start_admin_surveys_job(request):
-    # TODO: implement
-    pass
 
+def upload_group_polygons(request):
+    file_obj = request.FILES['file']
+    file_obj.seek(0)
+    text = file_obj.read()
+    data = json.loads(text)
 
-def admin_edits_page(request):
-    # TODO: implement
-    pass
+    group_data = OrderedDict([
+        (group.name, {
+            'group': group,
+            'previous_polygon_count':
+                len(group.border.coords) if group.border else 0,
+            'polygons': []})
+        for group in Group.objects.all()
+    ])
 
+    context = _update_group_polygons(data, group_data)
+    return context
 
-def admin_edits_partial(request):
-    # TODO: implement
-    pass
 
+@transaction.atomic
+def _update_group_polygons(data, group_data):
+    unknown_group_names = set()
 
-def admin_training_material_list(request):
-    # TODO: implement
-    pass
+    try:
+        for feature in data['features']:
+            group_name = feature['attributes']['GroupName']
+            polygon_points = feature['geometry']['rings'][0]
```

Comment:
Did we discuss, with Parks, the fact that we are not supporting polygons with holes? I don't want them to assume that it will work.